### PR TITLE
feat(stats): add tui output

### LINKED ISSUE #51: feat: make pb stats display in a tui
`pb stats` currently renders straight to stdout as colored text or JSON (`stats::run`/`run_with` in src/stats.rs, gated by the existing `--json` flag at src/cli.rs:108). The report (`StatsReport`: most-used, least-used, never-used, recency buckets, directory affinity — src/stats.rs:74-82) is assembled once and then handed to a text renderer.

Proposal: make a TUI the default presentation, reusing the existing `StatsReport` data, and keep the current text rendering reachable via a flag for scripting/piping — e.g. `pb stats --output text` (naming TBD; shouldn't collide with or duplicate the existing `--json`).

Implementation should follow patterns already established in `src/execute/`: a scrollable `Paragraph`/`List` (see src/execute/render.rs:151,254) showing the same sections as the current text report, respecting the `StdoutTtyGuard` terminal-restore invariants documented in CLAUDE.md on every exit path.

Open question: should `--json` continue to bypass the TUI entirely (it's meant for machine consumption), with the new output flag only toggling between TUI and the existing human-readable text rendering?

### LINKED ISSUE #54: refactor: make pb execute use shared TUI chrome
## Background

`src/tui/chrome.rs` provides shared TUI chrome used by newer screens like `pb new` and now `pb stats`, but its module docs explicitly note that execute still computes its own chrome:

> Execute itself does not (yet) call this; it computes its own bordered area with a custom two-pane header.

## Goal

Refactor `pb execute` rendering to use the shared `tui::chrome::Chrome` where practical so execute, new, and stats share consistent outer border/title/footer styling.

## Acceptance criteria

- `pb execute` uses the shared chrome helper (or the helper is evolved minimally so execute can use it without regressing behavior).
- Existing execute layout behavior remains intact: fuzzy/browse/tags modes, preview pane, prompt flow, status/help footer, and edit flow still work.
- Terminal invariants remain intact, especially the `StdoutTtyGuard`/raw-mode cleanup behavior for shell hotkey invocation.
- Existing execute tests pass; add/adjust focused rendering tests if the chrome boundary changes.

## Notes

Keep the change surgical. Avoid broad execute UI redesign beyond adopting the shared chrome styling.

### PR DESCRIPTION
**Pull request by calamity-m**
## Why

`pb stats` should default to an interactive TUI while preserving script-friendly output modes from the existing stats report.

Closes #51
Closes #54

## What

- Added `--output <tui|text|json>` for stats output, defaulting to `tui` for terminals and falling back to text when stdout is not a terminal.
- Removed the stats-specific `--json` flag in favor of `--output json`.
- Moved shared inline TUI terminal lifecycle helpers into `src/tui/terminal.rs` so stats and execute both depend on `tui` instead of stats depending on execute.
- Rendered the stats TUI with the shared `tui::chrome::Chrome` styling used by `pb new`, while preserving colored stats content.
- Refactored `pb execute` select/prompt rendering to use the shared `Chrome` border/title/footer while keeping the existing picker modes, preview pane, prompt flow, edit flow, and terminal safeguards.
- Added a shared scrollable inline text TUI helper that reuses execute terminal safeguards (`StdoutTtyGuard`, raw-mode RAII, event draining, viewport cleanup).
- Updated stats tests/examples to request `--output text` or `--output json` behavior explicitly.

## Verification

- `cargo fmt`
- `cargo test`
- `cargo clippy -- -D warnings`
- `cargo run -- stats --output text`
- `cargo run -- stats --output json`
- `cargo run -- stats --json` rejects the removed flag
- Zellij manual TUI check: launched `target/debug/peanutbutter stats`, confirmed shared chrome rendered, Down scrolled, and `q` exited/cleaned the viewport.

## Review focus

- Confirm the `--output <tui|text|json>` modes and removal of the stats-specific `--json` flag match expectations.
- Confirm the execute shared-chrome adoption is acceptable without broader UI redesign.

### CODE REVIEW

**Review by calamity-m — COMMENTED** [review]
## Review summary

Solid change overall — modelling output as a `Tui`/`Text`/`Json` enum is cleaner than the old boolean, reusing `Chrome` is the right call, and the `clap_rejects_removed_stats_json_flag` test is a nice guard. Two things worth addressing before merge: a module-placement smell, and a default-mode behaviour change. Details inline.

### Architectural: `run_scrollable_text` lives in the wrong module

`run_scrollable_text` has no execute-specific logic — it's a generic "render a string in a scrollable inline viewport with chrome" widget. It sits in `execute/terminal.rs` only because the terminal-lifecycle guards it needs (`StdoutTtyGuard`, `RawModeGuard`, `TuiOutputKind`, `build_terminal`, `cleanup_terminal`) are private there. The cost is a dependency pointing the wrong way: **`stats` now depends on `execute`** (`crate::execute::run_scrollable_text`).

The correct home already exists — `src/tui/`, whose `chrome.rs` doc literally says *"Adopt this helper for any new screen"*, and which `run_scrollable_text` already calls. The chrome was promoted to `tui`; the terminal guards + viewer were not. Proposed structure:

```text
src/tui.rs              -> module decl + re-exports (pub(crate))
src/tui/chrome.rs       -> Chrome (unchanged)
src/tui/terminal.rs     -> moved from execute/terminal.rs:
                             StdoutTtyGuard, RawModeGuard, TuiOutputKind,
                             build_terminal, cleanup_terminal,
                             run_scrollable_text
```

```rust
// src/tui.rs
pub(crate) mod chrome;
pub(crate) mod terminal;

pub(crate) use chrome::Chrome;
pub(crate) use terminal::run_scrollable_text;
// plus the guards, re-exported for execute's use
```

Then the dependency edges become:

```text
execute  ─┐
          ├─►  tui (chrome + terminal lifecycle + scrollable viewer)
stats    ─┘
```

`execute/terminal.rs` keeps `run_execute*` but consumes the guards from `tui` (same code path it has today — a move, not a rewrite). `stats` calls `crate::tui::run_scrollable_text`. The bad `stats → execute` edge disappears.

Note: CLAUDE.md flags the `StdoutTtyGuard` `dup2` pattern as load-bearing and "do not reimplement" — so this should be a literal *move* of those items, with the Unix path untouched.

File: src/execute/mod.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ pub use app::{
     AppEvent, ExecutionApp, NavigationMode, SuggestionProvider, SystemSuggestionProvider,
 };
 pub use prompt::{command_suggestions, render_command};
+pub(crate) use terminal::run_scrollable_text;
```
Comment (calamity-m) [inline src/execute/mod.rs]: **Dependency direction smell.** This re-export is what makes `stats` reach into `execute`. `run_scrollable_text` is a generic viewer with nothing execute-specific in it; it only lives here because the terminal-lifecycle guards it needs are private to this file. See the review summary for the proposed move into `src/tui/terminal.rs` so both `execute` and `stats` depend on `tui`, instead of `stats` depending on `execute`.

File: src/stats.rs [hunk 2]
```diff
@@ -18,23 +18,38 @@ pub enum Sort {
     Count,
 }
 
+/// Presentation mode for `peanutbutter stats`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, clap::ValueEnum)]
+pub enum Output {
+    /// Show statistics in an interactive terminal UI.
+    #[default]
+    Tui,
```
Comment (calamity-m) [inline src/stats.rs:26]: **Behaviour change — default flipped from text to interactive TUI, with no tty fallback.** Because `StdoutTtyGuard` redirects fd 1 to `/dev/tty`, `pb stats | cat`, `pb stats > file`, and any scripted use now launch an interactive TUI and write **nothing** to the pipe. Combined with the removal of `--json` (now `--output json`), existing scripts break. Suggest falling back to `Output::Text` when `!io::stdout().is_terminal()` — you already compute `is_terminal()` for the `color` decision in `run()`. At minimum this should be a conscious, documented choice.

File: src/execute/terminal.rs [hunk 3]
```diff
@@ -135,6 +141,70 @@ pub fn run_execute_with_provider<P: SuggestionProvider>(
...
+    let _stdout_guard = StdoutTtyGuard::enter()?;
+    let tui_output = TuiOutputKind::detect();
+    let _raw_mode = RawModeGuard::enter(tui_output)?;
+    let mut terminal = build_terminal(viewport_height, tui_output)?;
+    let mut viewport_top: Option<u16> = None;
+    let mut scroll = 0u16;
+    let text = text
+        .clone()
+        .into_text()
+        .unwrap_or_else(|_| Text::from(text));
```
Comment (calamity-m) [inline src/execute/terminal.rs]: **Nit:** the `.clone()` exists only to keep `text` alive for the fallback arm. `IntoText` works on `&str`, so this avoids the allocation:
```rust
let text = text.as_str().into_text().unwrap_or_else(|_| Text::from(text));
```

File: src/execute/terminal.rs [hunk 4]
```diff
@@ -135,6 +141,70 @@ pub fn run_execute_with_provider<P: SuggestionProvider>(
...
+    let line_count = text.lines.len() as u16;
+
+    loop {
+        terminal.draw(|frame| {
+            viewport_top = viewport_top.or(Some(frame.area().y));
+            let area = frame.area();
```
Comment (calamity-m) [inline src/execute/terminal.rs]: **Nit:** `frame.area()` is called twice on adjacent lines — fold into the single `let area` below and reuse it for the `viewport_top` init.

File: src/stats.rs [hunk 5]
```diff
@@ -316,6 +344,56 @@ fn write_json<W: Write>(writer: &mut W, report: &StatsReport) -> io::Result<()>
...
+    now: u64,
+    output: Output,
+    theme: &Theme,
+) -> io::Result<()> {
+    match output {
+        Output::Text => write_human(writer, report, sort, color, now),
+        Output::Json => write_json(writer, report),
+        Output::Tui => {
+            let mut rendered = Vec::new();
+            let tui_color = std::env::var_os("NO_COLOR").is_none();
```
Comment (calamity-m) [inline src/stats.rs:395]: The Tui arm forces colour on (`NO_COLOR.is_none()`), deliberately ignoring the `is_terminal()` gate the text path uses. Defensible (the TUI is always a terminal), but side-by-side with the `color` logic in `run()` it reads as contradictory — worth a one-line `// the TUI is always a terminal, so the is_terminal() gate does not apply` comment.

**Comment by calamity-m** [comment]
Addressed the review findings in f592b82:\n\n- moved shared inline TUI terminal lifecycle + scrollable viewer into src/tui/terminal.rs, removing the stats → execute dependency;\n- made default stats TUI output fall back to text when stdout is not a terminal;\n- actioned #54 in this PR by refactoring pb execute select/prompt rendering to use shared Chrome;\n- added focused tests for stats output fallback and execute shared chrome rendering.\n\nVerification: cargo test; cargo clippy -- -D warnings.
